controllers/profiles/reconciler_dev.go

```diff
@@ -280,55 +315,72 @@ func getDeploymentFailureMessage(deployment *appsv1.Deployment) string {
 }
 
 // mountDevConfigMapsMutateVisitor mounts the required configMaps in the Workflow Dev Deployment
-func mountDevConfigMapsMutateVisitor(flowDefCM, propsCM *v1.ConfigMap) mutateVisitor {
+func mountDevConfigMapsMutateVisitor(flowDefCM, propsCM *v1.ConfigMap, externalResourceConfigMaps map[string]v1.ConfigMap) mutateVisitor {
 	return func(object client.Object) controllerutil.MutateFn {
 		return func() error {
 			deployment := object.(*appsv1.Deployment)
 			volumes := make([]v1.Volume, 0)
 			volumeMounts := make([]v1.VolumeMount, 0)
 
 			volumes = append(volumes,
-				v1.Volume{
-					Name: configMapWorkflowDefVolumeName,
-					VolumeSource: v1.VolumeSource{
-						ConfigMap: &v1.ConfigMapVolumeSource{
-							LocalObjectReference: v1.LocalObjectReference{Name: flowDefCM.Name},
-						},
-					},
-				},
-				v1.Volume{
-					Name: configMapWorkflowPropsVolumeName,
-					VolumeSource: v1.VolumeSource{
-						ConfigMap: &v1.ConfigMapVolumeSource{
-							LocalObjectReference: v1.LocalObjectReference{Name: propsCM.Name},
-							Items:                []v1.KeyToPath{{Key: applicationPropertiesFileName, Path: applicationPropertiesFileName}},
-						},
-					},
-				})
+				kubeutil.Volume(configMapWorkflowDefVolumeName, flowDefCM.Name),
+				kubeutil.VolumeWithItems(configMapWorkflowPropsVolumeName, propsCM.Name, []v1.KeyToPath{{Key: applicationPropertiesFileName, Path: applicationPropertiesFileName}}),
+			)
 
 			volumeMounts = append(volumeMounts,
-				v1.VolumeMount{
-					Name:      configMapWorkflowDefVolumeName,
-					ReadOnly:  true,
-					MountPath: configMapWorkflowDefMountPath,
-				},
-				v1.VolumeMount{
-					Name:      configMapWorkflowPropsVolumeName,
-					ReadOnly:  true,
-					MountPath: quarkusDevConfigMountPath,
-				},
+				kubeutil.VolumeMount(configMapWorkflowDefVolumeName, true, configMountPathMap[WORKFLOW]),
+				kubeutil.VolumeMount(configMapWorkflowPropsVolumeName, true, quarkusDevConfigMountPath),
 			)
 
+			externalVolumes, externalVolumesMount := kubeutil.ConfigMapsToVolumesAndMount(externalResourceConfigMaps, prefixMountPathMap)
+			volumes = append(volumes, externalVolumes...)
+			volumeMounts = append(volumeMounts, externalVolumesMount...)
+
 			deployment.Spec.Template.Spec.Volumes = make([]v1.Volume, 0)
 			deployment.Spec.Template.Spec.Volumes = volumes
 			deployment.Spec.Template.Spec.Containers[0].VolumeMounts = make([]v1.VolumeMount, 0)
 			deployment.Spec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
-
 			return nil
 		}
 	}
 }
 
+func (e ensureRunningDevWorkflowReconciliationState) fetchExternalResourcesConfigMaps(workflow *operatorapi.KogitoServerlessWorkflow) map[string]v1.ConfigMap {
+	externalConfigMaps := make(map[string]v1.ConfigMap, 0)
+	for k, val := range workflow.Labels {
```

Comment:
I don't think that the **_labels_** is the right place where to put this info because it is usually used for a different purpose (see: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)

Reply:
Alternatively, instead o labels we use annotations to the CM.

Reply:
Just use `annotations` instead.

Reply:
ok

Reply:
@ricardozanini labels turned into annotations